### PR TITLE
Fix up field attrs (fixes #137)

### LINKED ISSUE #137: Ensure name attrs aren't specified on the form
This is so the form values aren't submitted to us if the js submit breaks.
Needs test coverage too.

### PR DESCRIPTION
**Pull request by muffinresearch**
* Remove the braintree data attrs as we're building things up at a lower level.
* Add the autocomplete="off" attr
* Add test for no name attr being present.
* Test for type="tel" and autocomplete

### CODE REVIEW

File: public/jsx/components/card-form.jsx [hunk 1]
```diff
@@ -61,18 +69,17 @@ module.exports = React.createClass({
...
       'pattern': '11/11',
       'placeholder': 'MM/YY',
-      'type': 'tel',
       'validator': CardValidator.expirationDate,
     },
     cvv: {
+      'attrs': defaultFieldAttrs,
+      'autocomplete': 'off',
       'classNames': ['cvv'],
-      'data-braintree-name': 'cvv',
```
Comment (andymckay) [inline public/jsx/components/card-form.jsx]: braintree doesn't need this anymore, or is it added somewhere else?

Comment (andymckay) [inline public/jsx/components/card-form.jsx]: oh i guess that's for drop-in ui?

**Comment by andymckay** [comment]
r+wc

File: public/jsx/components/card-form.jsx (hunk 1 above)
Comment (muffinresearch) [inline public/jsx/components/card-form.jsx]: Yep - we do the tokenisation at a lower level.

**Pull request merged**